File: haskell/app/WordCount.hs
module Main where

import DataProcessing.WordCount (countWords)
import qualified Data.HashMap.Strict as HM
import Options.Applicative

newtype Args = Args { argInput :: FilePath }

argsParser :: ParserInfo Args
argsParser = info (Args <$> strOption (long "input" <> metavar "FILE") <**> helper) mempty

main :: IO ()
main = do
    Args path <- execParser argsParser
    counts <- countWords path
    let totalWords = sum (HM.elems counts)
    putStrLn $ "unique=" ++ show (HM.size counts) ++ " total=" ++ show totalWords

File: rust/src/bin/word_count.rs
use clap::Parser;
use fp_benchmarks::data_processing::word_count;
use std::path::PathBuf;

#[derive(Parser)]
struct Args {
    #[arg(long)]
    input: PathBuf,
}

fn main() {
    let args = Args::parse();
    let counts = word_count::count_words(&args.input);
    // Output total unique words and total count
    let total_words: u64 = counts.values().sum();
    println!("unique={} total={total_words}", counts.len());
}
